lib/Stats.tsx

```diff
@@ -1,101 +1,236 @@
   position: absolute;
   top: 24px;
   left: 24px;
-  padding: 8px 16px;
-  font-family: sans-serif;
+  width: 360px;
+  min-width: 240px;
   max-width: 80%;
   max-height: 80%;
+  border-radius: 4px;
+  background: #292929 0% 0% no-repeat padding-box;
+  opacity: 0.88;
+  overflow: scroll;
 `
 
 const StatsHeader = styled.div`
-  color: white;
+  padding: 8px 24px;
+  border-bottom: 1px solid #525252;
+`
+
+const StatsIcon = styled.span<{ readonly clickable: boolean }>`
+  width: 24px;
+  height: 24px;
+  float: left;
+  cursor: ${({ clickable }) => (clickable ? 'pointer' : 'default')};
+
+  & > svg {
+    fill: #e0e0e0;
+  }
+`
+
+const StatsTitle = styled.span`
+  display: inline-block;
+  margin-left: 8px;
+  vertical-align: sub;
   text-align: left;
-  font-size: 14px;
-  font-weight: bold;
+  font-size: 16px;
+  font-family: 'Open Sans';
+  line-height: 22px;
+  color: #f5f5f5;
 `
 
-const StatContainer = styled.div`
-  display: flex;
-  flex-direction: row;
-  flex-wrap: wrap;
-  justify-content: left;
+const StatsHide = styled.span`
+  float: right;
+  text-align: right;
 `
 
-const Stat = styled.div`
-  color: white;
+const HideLink = styled.a`
+  vertical-align: sub;
+  text-decoration: none;
+  font-size: 16px;
+  font-family: 'Open Sans';
+  line-height: 22px;
+  color: #b8b8b8;
+`
+
+const StatsData = styled.div`
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  column-gap: 24px;
+  row-gap: 16px;
+  width: 100%;
+  padding: 16px 24px 24px;
+`
+
+const StatItem = styled.div`
   text-align: left;
-  margin: 8px 0 0;
-  width: 45%;
+  font-family: 'Open Sans';
+`
+
+const StatName = styled.div`
   font-size: 12px;
+  line-height: 17px;
+  color: #b8b8b8;
+`
+
+const StatValue = styled.div`
+  font-size: 13px;
+  line-height: 18px;
+  color: #e0e0e0;
 `
 
-const Title = styled.div`
-  font-weight: bold;
+const StatsShow = styled.div`
+  position: absolute;
+  top: 24px;
+  left: 24px;
 `
 
 interface StatsProps {
   readonly api: string
   readonly parameters: VapixParameters
   readonly videoProperties: VideoProperties
-  readonly host: string
   readonly open: boolean
   readonly refresh: number
 }
 
+interface Stat {
+  readonly name: string
+  readonly value: string | number
+  readonly unit?: string
+}
+
 export const Stats: React.FC<StatsProps> = ({
   api,
   parameters,
   videoProperties,
-  host,
   refresh,
 }) => {
-  const streamType = useMemo(() => {
-    if (api === 'jpg') {
-      return 'Still image'
-    }
+  const [stats, setStats] = useState<Array<Stat>>([])
+  const [showStats, setShowStats] = useState(true)
 
-    if (api === 'media') {
-      if (parameters['videocodec'] === 'h264') {
-        return 'H.264'
+  // Handles show/hide stats
+  const onToggleStats = useCallback(
+    (e) => {
+      e.preventDefault()
+      setShowStats((prevState) => !prevState)
+    },
+    [setShowStats],
+  )
+
+  // Updates stat values
+  const statsUpdateRef = useRef(0)
+  useEffect(() => {
```

Comment:
If this only affects the `StatsData` component, it should probably be in that component.

Reply:
Done